std/node/_fs/_fs_appendFile.ts

```diff
@@ -23,37 +23,48 @@ export async function appendFile(
   validateEncoding(options);
 
   let rid = -1;
-  try {
-    if (typeof pathOrRid === "number") {
-      rid = pathOrRid;
-    } else {
-      const mode: number | undefined = isFileOptions(options)
-        ? options.mode
-        : undefined;
-      const flag: string | undefined = isFileOptions(options)
-        ? options.flag
-        : undefined;
-
-      if (mode) {
-        //TODO rework once https://github.com/denoland/deno/issues/4017 completes
-        notImplemented("Deno does not yet support setting mode on create");
+  new Promise(async (resolve, reject) => {
+    try {
+      if (typeof pathOrRid === "number") {
+        rid = pathOrRid;
+      } else {
+        const mode: number | undefined = isFileOptions(options)
+          ? options.mode
+          : undefined;
+        const flag: string | undefined = isFileOptions(options)
+          ? options.flag
+          : undefined;
+
+        if (mode) {
+          //TODO rework once https://github.com/denoland/deno/issues/4017 completes
+          notImplemented("Deno does not yet support setting mode on create");
```

Comment:
It seems odd that this is inside the try block... It seems like it would be better if only Deno.open was in the try block.